docker/docker-compose.yml

```diff
@@ -1,31 +1,94 @@
-version: "3.7"
+version: "3.8"
 services:
-  minio:
-    image: minio/minio # Stable minio
+  coordinator:
+    image: xain/xain-fl:development
+    build:
+      context: ..
+      dockerfile: docker/Dockerfile
+    depends_on:
+      - minio
+      - redis
+      - influxdb
+    command: /app/coordinator -c /app/config.toml
     volumes:
-      - minio-data:/data
+      - ${PWD}/../configs/docker-dev-coordinator.toml:/app/config.toml
     networks:
       - xain-fl
+    ports:
+      - "5555:5555"
+      - "8081:8081"
+
+  influxdb:
+    image: influxdb:1.8
+    hostname: influxdb
+    container_name: influxdb
     environment:
-      MINIO_ACCESS_KEY: minio
-      MINIO_SECRET_KEY: minio123
+      INFLUXDB_DB: metrics
+      INFLUXDB_DATA_QUERY_LOG_ENABLED: 'false'
+      INFLUXDB_HTTP_LOG_ENABLED: 'false'
+    volumes:
+      - influxdb-data:/var/lib/influxdb
+    networks:
+      - xain-fl
+    expose:
+      - "8086"
+
+  minio:
+    image: minio/minio
+    hostname: minio
+    container_name: minio
+    env_file:
+      - .dev.env
     command: server /data
+    volumes:
+      - minio-data:/data
+    networks:
+      - xain-fl
     ports:
       - "9000:9000"
 
+  setup-initial-buckets:
+    image: minio/mc
+    hostname: setup-initial-buckets
+    container_name: setup-initial-buckets
+    depends_on:
+      - minio
+    env_file:
+      - .dev.env
+    entrypoint: >
+      /bin/sh -c "
+        echo -n 'Waiting for minio service to start...';
+        while ! nc -z minio 9000;
+        do
+          sleep 1;
+          echo -n ".";
+        done;
+        echo ' Connected!';
+        mc config host add minio http://minio:9000 $${MINIO_ACCESS_KEY} $${MINIO_SECRET_KEY};
+        mc config host add minio http://minio:9000 $${MINIO_ACCESS_KEY} $${MINIO_SECRET_KEY};
+        /usr/bin/mc mb -p minio/xain-fl;
+        /usr/bin/mc policy set upload minio/xain-fl;
+        /usr/bin/mc policy set download minio/xain-fl;
+      "
+    networks:
+      - xain-fl
+
   redis:
-    image: redis:6.0.2 # Stable redis
+    image: redis:6
     hostname: redis
+    container_name: redis
+    entrypoint: /usr/local/bin/redis-server --appendonly yes --appendfsync everysec # using combination of RDB and AOF for persistence: https://redis.io/topics/persistence
     volumes:
       - redis-data:/data
     networks:
       - xain-fl
-    ports:
-      - "6379:6379"
+    expose:
```

Comment:
I would like to keep `ports` here because sometimes I need to access Redis via the redis-cli.

Reply:
I had changed this when we were running the dev `docker-compose` stack on DigitalOcean for the first testing sessions and we ended up with an inadvertently exposed Redis.
Can revert back to `ports`, as this docker-compose shouldn't be used in production anyway.